client/app/bundles/course/assessment/submission/reducers/form.js

```diff
@@ -22,6 +22,26 @@ function updateAnswerInState(state, answer) {
   }, { [id]: answer.fields });
 }
 
+function removeAnswerFileFromState(state, questionId, fileId) {
```

Comment:
Suggest a rename to `removeProgrammingFileFromState` - because this might be confused with file response questions.